lib/Doctrine/ORM/Internal/Hydration/SimpleObjectHydrator.php

```diff
@@ -151,4 +155,23 @@ protected function hydrateRowData(array $sqlResult, array &$result)
             $this->_uow->hydrationComplete();
         }
     }
+
+    /**
+     * @param string $discriminator
+     * @param array $sqlResult
+     */
+    private function removeColumnsNotRelatedToEntityDiscriminator(string $discriminator, array &$sqlResult)
+    {
+        foreach ($sqlResult as $column => $value) {
+            $cacheKeyInfo = $this->hydrateColumnInfo($column);
+
+            if (!isset($cacheKeyInfo['discriminatorValue'])) {
+                continue;
+            }
+
+            if ($cacheKeyInfo['discriminatorValue'] !== $discriminator) {
+                unset($sqlResult[$column]);
```

Comment:
Is removing the columns the correct approach here? Where is the hydration to DBAL type happening, precisely? Can we skip there instead? (we'd even win some performance, I think)

Reply:
Yeah, I think the correct approach is filtering upfront, not once the row is being processed.

Reply:
Look, I need to know what correct discriminator. Correct discriminator contains in `$sqlResult`, but in `hydrateColumnInfo` method I have only column name. I dont have sql result set( I cant determinate correct discriminator.